libcasr/src/csharp.rs

```diff
@@ -0,0 +1,128 @@
+                stentry.address = address;
+            } else {
+                return Err(Error::Casr(format!("Couldn't parse address: {num}")));
+            }
+        }
+
+        stentry.function = cap.get(1).unwrap().as_str().to_string();
+
+        Ok(stentry)
+    }
+}
+
+impl ParseStacktrace for CSharpStacktrace {
+    fn extract_stacktrace(stream: &str) -> Result<Vec<String>> {
+        let re = Regex::new(r"(?m)^Unhandled ([Ee])xception(:\n|\. )(?:.|\n)+?((?:\n[^\S\n]*(?:at.+|--- End of inner exception stack trace ---))+)$").unwrap();
+
+        let Some(cap) = re.captures(stream).and_then(|cap|
+            ((cap.get(1).unwrap().as_str() == "E") == (cap.get(2).unwrap().as_str() == ":\n")).then_some(cap)
+        ) else {
+            return Err(Error::Casr("Couldn't find stacktrace".to_string()));
+        };
+
+        Ok(cap
+            .get(3)
+            .unwrap()
+            .as_str()
+            .split('\n')
+            .map(|s| s.trim().to_string())
+            .filter(|s| !s.is_empty())
+            .collect::<Vec<String>>())
+    }
+
+    fn parse_stacktrace_entry(entry: &str) -> Result<StacktraceEntry> {
+        match Self::parse_stacktrace_entry(entry, true) {
+            Err(Error::Casr(ref s)) if s.starts_with(PARSE_STACK_FRAME_ERR_PREFIX) => Self::parse_stacktrace_entry(entry, false),
+            r @ _ => r
+        }
+    }
+
+    fn parse_stacktrace(entries: &[String]) -> Result<Stacktrace> {
+        let iter = entries
+            .iter()
+            .scan((true, false), |(cancel_skip, skip), s| {
+                // Skipping all blocks consisting of "--- End of inner exception stack trace ---"
+                // and one stack frame after each such block, except for the first block if entries start with it.
+                let not_stack_frame = s == "--- End of inner exception stack trace ---";
+
+                if not_stack_frame || *skip {
+                    *skip = not_stack_frame;
+
+                    if !*cancel_skip || not_stack_frame {
+                        return Some("");
+                    }
+                }
+
+                *cancel_skip = false;
+
+                Some(s)
+            })
+            .filter(|&s| !s.is_empty());
+
+        let re_mono = Regex::new(r"^at (?:\(.+\) )?(.+?) ?\(.*\) (?:<0x\w+.+>|\[\w+]) in (?:<\w+>|.+):\w+$").unwrap();
+        let use_mono_format = iter.clone().all(|s| re_mono.captures(s).is_some());
+
+        iter
+            .map(|s| Self::parse_stacktrace_entry(s, use_mono_format))
+            .collect()
+    }
+}
+
+/// Structure provides an interface for parsing c sharp exception message.
+pub struct CSharpException;
+
+impl Exception for CSharpException {
+    fn parse_exception(stream: &str) -> Option<ExecutionClass> {
+        let re_mono = Regex::new(r"(?m)^Unhandled Exception:\n((?:.|\n)+?(?: ---> (?:.|\n)+?)*?)\n[^\S\n]*(at.+|--- End of inner exception stack trace ---)$").unwrap();
```

Comment:
Maybe `Unhandled Exception:\n((?:.|\n)+?` -> `Unhandled Exception:\n((?:.|\n)*?`  
I think the part `(?:.|\n)` may not be present.